### ISSUE TITLE
[RLLib] Dueling DQN doesn't work with masked actions

### ISSUE THREAD

**Issue by ingambe** [ray-project/ray#14355]
### What is the problem?

Currently, Dueling DQN is not working with the masked actions.
I've started to work on it. See the related PR.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray 2.0
Pytorch 1.7.1
macOS Catalina

### Reproduction (REQUIRED)
Use Parametric Cartpole or an environment with masked actions (I use: https://pypi.org/project/JSSEnv/ and https://github.com/prosysscience/JSS/blob/master/JSS/models.py as a model). Then activate DQN with Dueling.
This is a known issue as it is reported in the Parametric Cartpole example (but not in the doc).

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).
